src/porepy/numerics/nonlinear/nonlinear_solvers.py

```diff
@@ -36,6 +36,7 @@ def __init__(self, params=None) -> None:
         default_options = {
             "max_iterations": 10,
             "nl_convergence_tol": 1e-10,
+            "nl_convergence_tol_res": np.inf,
```

Comment:
Why this tolerance?

Reply:
The default case was intended to not break with the previous state of the code. This tolerance deactivates checking residuals to accept convergence. With this, the solver behavior of existing simulations should not change with this PR.